### PR TITLE
update docs for clarity and easier comprehension for new users

### PR DESCRIPTION
**Pull request by TheeMattOliver**
Thanks for building changesets! Just proposing a few changes to the docs that fix some minor grammatical errors and made a few proposed changes that might improve readability and comprehension for new users. Some of the wording in the docs on snapshot releases was kind of hard to parse, at least as a first-time user coming to the docs and learning the tool. Anyway, please feel free to just grab any changes you find helpful and close this if that's easier. Thanks! ✌️

### CODE REVIEW

**Comment by Andarist** [comment]
Thank you very much 🙏 The doc PRs are the best ones! Would you maybe be so kind to add a note about [this newly merged in feature](https://github.com/changesets/changesets/commit/b9b6453b16ada9751e2280b5230ded72161391e9#diff-32d8f2bc534881208052c344ae49d59f2b0daf12ac8de612afb745395d7e8772R5)? Using this flag might be important when publishing the snapshot releases locally or when for some reason git tags are meant to be pushed to the remote from your CI env (well, they should be pushed when releasing stable versions, but probably they shouldn't be for snapshot releases)

**Comment by TheeMattOliver** [comment]
Happy to do it. I think [this captures the use case](https://github.com/changesets/changesets/pull/717/commits/aa9031657b405fe9c8bfb46a491f666e91593e92), but feel free to reject if it still needs work :) 

One more note -- I merged in some changes from the remote so this branch was up to date with main, but not sure if I should have or not 🙈 .

File: docs/snapshot-releases.md [hunk 1]
```diff
@@ -24,7 +24,13 @@ This will instead update versions to `0.0.0-bulbasaur-THE_TIME_YOU_DID_THIS`
 
 ## Publishing your packages
 
-After running version, you can use the `changeset publish --tag bulbasaur` command to releases the packages. By using the `--tag` flag, you will not add it to the `latest` flag on npm. This is REALLY IMPORTANT because if you do not include a tag, people installing your package using `yarn add your-package-name` will install the snapshot version.
+After running the `yarn changeset version` command, you can use the `changeset publish --tag bulbasaur` command to releases the packages. By using the `--tag` flag, you will not add it to the `latest` flag on npm. This is REALLY IMPORTANT because if you do not include a tag, people installing your package using `yarn add your-package-name` will install the snapshot version.
+
+## Using the `--no-git-tag` flag
+
+You can use the `--no-git-tag` CLI flag when running `changeset publish` if you plan to publish snapshot releases locally and you are pushing [git tags](http://npm.github.io/publishing-pkgs-docs/updating/using-tags.html) to a remote from your CI environment.
```
Comment (Andarist) [inline docs/snapshot-releases.md:31]:
```suggestion
You can use the `--no-git-tag` CLI flag when running `changeset publish` if you plan to publish snapshot releases locally or you are pushing [git tags](http://npm.github.io/publishing-pkgs-docs/updating/using-tags.html) to a remote from your CI environment. 
```

File: docs/snapshot-releases.md [hunk 2]
```diff
@@ -24,7 +24,13 @@ This will instead update versions to `0.0.0-bulbasaur-THE_TIME_YOU_DID_THIS`
...
+After running the `yarn changeset version` command, you can use the `changeset publish --tag bulbasaur` command to releases the packages. By using the `--tag` flag, you will not add it to the `latest` flag on npm. This is REALLY IMPORTANT because if you do not include a tag, people installing your package using `yarn add your-package-name` will install the snapshot version.
+
+## Using the `--no-git-tag` flag
+
+You can use the `--no-git-tag` CLI flag when running `changeset publish` if you plan to publish snapshot releases locally and you are pushing [git tags](http://npm.github.io/publishing-pkgs-docs/updating/using-tags.html) to a remote from your CI environment. 
+
+When you run `changeset publish --no-git-gag`, changesets will skip creating git tags for published packages. That means that git tags can be still be created whenever pushing stable versions, and you can safely publish snapshot releases locally, without creating unnecessary tags.
```
Comment (Andarist) [inline docs/snapshot-releases.md:33]:
```suggestion
When you run `changeset publish --no-git-tag --snapshot`, changesets will skip creating git tags for published snapshot packages. That means that git tags can still be created whenever pushing stable versions (with a regular `changeset publish`), and you can safely publish snapshot releases locally, without creating unnecessary tags.
```

**Comment by Andarist** [comment]
> One more note -- I merged in some changes from the remote so this branch was up to date with main, but not sure if I should have or not 🙈 .

That's totally fine - I'm going to squash this at the end anyway.

I've left 2 minor suggestions, let me know what do you think about them.
